### PR TITLE
Minor adjustments to Option's doc

### PR DESCRIPTION
**Pull request by xavierguihot**

### CODE REVIEW

**Comment by SethTisue** [comment]
review by @ashawley?

File: src/library/scala/Option.scala [hunk 1]
```diff
@@ -509,7 +509,7 @@ sealed abstract class Option[+A] extends Product with Serializable {
     *
     *  This is equivalent to:
     *  {{{
-    *  option.unzip match {
+    *  option match {
```
Comment (ashawley) [inline src/library/scala/Option.scala:512]: Good catch!  Yeah, this is inconsistent with the rest of the examples.

File: src/library/scala/Option.scala [hunk 2]
```diff
@@ -534,7 +534,7 @@ sealed abstract class Option[+A] extends Product with Serializable {
     *
     *  This is equivalent to:
     *  {{{
-    *  option.unzip3 match {
+    *  option match {
```
Comment (ashawley) [inline src/library/scala/Option.scala:537]: Same here.  Thanks for fixing this.

File: src/library/scala/Option.scala [hunk 3]
```diff
@@ -484,13 +484,13 @@ sealed abstract class Option[+A] extends Product with Serializable {
...
    *  {{{
-   *  option1.zip(option2) match {
+   *  (option1, option2) match {
    *    case (Some(x), Some(y)) => Some((x, y))
    *    case _                  => None
    *  }
    *  }}}
    *  @example {{{
-   *  // Returns Option(("foo", "bar")) because both options are nonempty.
+   *  // Returns Some(("foo", "bar")) because both options are nonempty.
```
Comment (ashawley) [inline src/library/scala/Option.scala:493]: Yes, it should be `Some`, not `Option`.

File: src/library/scala/Option.scala [hunk 4]
```diff
@@ -484,13 +484,13 @@ sealed abstract class Option[+A] extends Product with Serializable {
    *
    *  This is equivalent to:
    *  {{{
-   *  option1.zip(option2) match {
+   *  (option1, option2) match {
```
Comment (ashawley) [inline src/library/scala/Option.scala:487]: LGTM.

**Review by ashawley — APPROVED** [review]
Looks good.  This fixes my mistakes in #7727.  Thank you.

**Review by diesalbla — APPROVED** [review]

**Pull request merged**